src/plugins/category/plugin.js

```diff
@@ -0,0 +1,470 @@
+            {{?}} \
+          {{?}} \
+          <option value="{{= category.id }}">{{= it.translate(category.label) }}</option> \
+        {{~}} \
+        {{? optgroup !== null }}</optgroup>{{?}} \
+      </select>'
+  }
+});
+
+QueryBuilder.define('category', function(options) {
+    var self = this;
+
+    // SETTINGS SHORTCUT
+    this.categories = this.settings.categories;
+
+    // CHECK ALL CATEGORIES
+    this.categories = this.checkCategories(this.categories);
+    /**
+     * Bind events
+     */
+    this.on('afterInit', function() {
+        // change event bindind for rule category select box
+        self.$el.on('change.queryBuilder', Selectors.rule_category, function() {
+            var $rule = $(this).closest(Selectors.rule_container);
+            Model($rule).category = self.getCategoryById($(this).val());
+        });
+
+        // update event binding for category field of rule model 
+        self.model.on('update', function(e, node, field, value, oldValue) {
+            if (node instanceof Rule && field === 'category') {
+                self.updateRuleCategory(node, oldValue);
+            }
+        });
+    });
+
+    /**
+     * Initialize category property of a rule and create rule categories select box when new rule is added
+     */
+    this.on('afterAddRule', function(e, rule) {
+        rule.__.category = null;
+        self.createRuleCategories(rule);
+    });
+
+    /**
+     * Modify templates
+     */
+    this.on('getRuleTemplate.filter', function(h) {
+        var $h = $(h.value);
+        $h.find(Selectors.filter_container).before('<div class="rule-category-container"></div>');
+        h.value = $h.prop('outerHTML');
+    });
+
+    // init selectpicker
+    this.on('afterCreateRuleCategories', function(e, rule) {
+        if (self.settings.plugins && self.settings.plugins['bt-selectpicker']) {
+            self.$el.find(Selectors.rule_category).removeClass('form-control').selectpicker(options);
+        }
+    });
+
+    // update selectpicker on change
+    this.on('afterUpdateRuleCategory', function(e, rule) {
+        if (self.settings.plugins && self.settings.plugins['bt-selectpicker']) {
+            self.$el.find(Selectors.rule_category).selectpicker('render');
+        }
+    });
+
+    /**
+     * Export "category" to JSON
+     */
+    this.on('ruleToJson.filter', function(e, rule) {
+        e.value.category = rule.category.id;
+    });
+
+    /**
+     * Read "category" from JSON
+     */
+    this.on('jsonToRule.filter', function(e, json) {
+        //e.value.category = e.builder.getCategoryById(json.category ? json.category : '-1');
+    });
+}, {
+    default_no_category: false,
+    container: 'body',
+    style: 'btn-inverse btn-xs',
+    width: 'auto',
+    showIcon: false
+});
+
+// keep a pointer to the original addRule method
+var originaAddRule = QueryBuilder.prototype.addRule;
+
+QueryBuilder.extend({
```

Comment:
Not ok, you must find a way the call the original method (but putting the new code before or after) or use existing events, or add new events where needed.